src/io/flutter/sdk/FlutterSdkUtil.java

```diff
@@ -163,6 +166,23 @@ private static String readVersionFile(String sdkHomePath) {
     return null;
   }
 
+  public static boolean isFlutterModule(@Nullable Module module) {
+    return module != null && ModuleType.is(module, FlutterModuleType.getInstance());
+  }
+
+  public static boolean hasFlutterModule(@NotNull Project project) {
+    return ModuleUtil.hasModulesOfType(project, FlutterModuleType.getInstance());
+  }
+
+  /**
+   * Checks the workspace for any open Flutter projects.
+   *
+   * @return true if an open Flutter project is found
+   */
+  public static boolean isFluttering() {
+    return Arrays.stream(ProjectManager.getInstance().getOpenProjects()).anyMatch((Project p) -> FlutterSdkUtil.hasFlutterModule(p));
```

Comment:
Or, better, delete `isFluttering()` method and call `hasFlutterModule(p)` right from the `DeviceSelectorAction.update()`.
It is very unusual when the presence of some other open project affects UI or behavior of the currently focused one.